specs/decimal/decimal.md

```diff
@@ -197,66 +339,307 @@ Windows App SDK provides a native language decimal data type for C++ as the
-   6. Rename Mul() to Multiply()
-   7. Rename Div() to Divide()
-   8. Rename Mod() to Remainder()
-
-# 6.2. TODO
-
-Punchlist to reach Stable:
-
-1. C++ class
-   1. BUG? decimal zero{}; decimal neg{ -zero }; neg.sign() < 0 because DECIMAL.sign = 0x80. Treat -0 the same as +0 (prevent -0 from being set?)
-2. winrt::Microsoft::Windows::Foundation::Decimal
-   1. Implement IInspectable methods -- Create(IInspectable), Set(IInspectable), ToObject()
-   2. Add experimental checks in ctor + static methods
-3. decimalcppwinrt.h
-   1. Merge into decimal.h
-4. Microsoft.Windows.Foundation.Projection
-   1. --none--
-5. C# Tests
-   1. TAEF for C# ?
-   2. Port test\inc\WindowsAppRuntime.Test.Package.h to C#
-   3. Port test\inc\WindowsAppRuntime.Test.Bootstrap.h to C#
-   4. Register framework package
-   5. Bootstrap.Initialize()
-   6. Implement remaining tests
+## 8.1. decimal.h
+
+```c++
+#if !defined(__WindowsAppSDK_Microsoft_Windows_Foundation_decimal_)
+#define __WindowsAppSDK_Microsoft_Windows_Foundation_decimal_
+
+#include <oleauto.h>
+
+#if !defined(_VC_NODEFAULTLIB)
+#pragma comment(linker, "/defaultlib:oleaut32.lib")
+#endif
+
+inline int compare(const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator==(const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator!=(const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator< (const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator<=(const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator> (const DECIMAL& left, const DECIMAL& right);
+
+inline bool operator>=(const DECIMAL& left, const DECIMAL& right);
+
+namespace Microsoft::Windows::Foundation
+{
+class decimal
+{
+public:
+    decimal() = default;
+    ~decimal() = default;
+
+    decimal(const decimal& value);
+    decimal(decimal&& value);
+    constexpr decimal(const DECIMAL& value);
+    decimal(bool value)l
+    decimal(char value);
+    decimal(std::int16_t value);
+    decimal(std::int32_t value);
+    decimal(std::int64_t value);
+    decimal(std::uint8_t value);
+    decimal(std::uint16_t value);
+    decimal(std::uint32_t value);
+    decimal(std::uint64_t value);
+    decimal(float value);
+    decimal(double value);
+    decimal(long value);
+    decimal(unsigned long value);
+    decimal(PCWSTR value);
+    decimal(PCWSTR value, const LCID locale);
+    decimal(const std::wstring& value);
+    decimal(const std::wstring& value, const LCID locale);
+#if defined(__WINSTRING_H_)
+    decimal(const HSTRING& value);
+    decimal(const HSTRING& value, const LCID locale);
+#endif // defined(__WINSTRING_H_)
+#if defined(WINRT_BASE_H)
+    decimal(const winrt::hstring& value);
+    decimal(const winrt::hstring& value, const LCID locale);
+#endif // defined(WINRT_BASE_H)
+
+    decimal& operator=(const decimal& value);
+    decimal& operator=(decimal&& value);
+    decimal& operator=(const DECIMAL& value);
+    decimal& operator=(bool value);
+    decimal& operator=(char value);
```

Comment:
Similar issues here -- do we really want the following to compile?

dec = 'A';
dec = false;
dec = L"Hello";

Reply:
RECOMMEND: Replace oper=(string) with explicit named methods, drop to/from(char) and consider dropping to/from(bool) per previous comment